include/deal.II/fe/fe_values.h

```diff
@@ -292,6 +316,13 @@ namespace FEValuesViews
                               std::vector<typename ProductType<value_type,typename InputVector::value_type>::type> &values) const;
 
     /**
+     *
+     */
+    template <class InputVector>
+    void get_function_values_from_local_dof_values (const InputVector &dof_values,
+                                                    std::vector<typename OutputType<typename InputVector::value_type>::value_type> &values) const;
```

Comment:
Yes, _all_ of the documentation is missing! We're first sketching this out, and will work on the tests and fill out the documentation this morning.

You'll notice that `OutputType::value_type` involves a `typename std::remove_cv<Number>::type` (necessary due to a const `Number` being passed to the function at some stage). We thought it would be easier for a user to write
```c++
std::vector<typename OutputType<Number>::value_type> qp_values (n_q_points);
```
and then
```c++
fe_values[sclr_extrcr].get_function_values_from_local_dof_values(local_dof_values, qp_values);
```
, rather than to try to work out the correct number type themselves.

To further facilitate this, the `OutputType` struct names all types returned by the `get_function_*_from_local_dof_values` functions, and is different for each `FEValuesView` type. 

We presumed that the `FEValuesView` classes are not typedef'd on `Number` as, presumably, the `value_type` etc. refer to the basis functions. We decided it would be unreasonable to typedef the whole class on `Number`.